### ISSUE TITLE
Different Metrics

### ISSUE THREAD

**Issue by sinagh72** [adap/flower#1845]
### What is your question?

I recently worked on training a ResNet-18 model for binary image classification using PyTorch lightning with the FedAvg strategy simulation. I had 3 clients and rounds = 10. However, I noticed that the results (AUC) varied across different runs, despite setting the torch seed, PyTorch lightning seed, and numpy seed:
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False
    torch.manual_seed(42)
    pl.seed_everything(42)
    np.random.seed(seed=42)

Is there anything else I need to set to get the same AUC each time on my test set?

**Comment by adam-narozniak** [comment]
Hi, could you set the seed for the Python random library as below:
```python
import random

random.seed(10)
```
The only part of Flower that I can think of that introduces randomness is sampling in client_manager (by default a `SimpleClientManager` is created).
https://github.com/adap/flower/blob/60a7230ab3f60b1ba56b5232718472e44c97037e/src/py/flwr/server/client_manager.py#L170-L199
Line 198 exactly:
https://github.com/adap/flower/blob/60a7230ab3f60b1ba56b5232718472e44c97037e/src/py/flwr/server/client_manager.py#L198

Let me know if that help.

**Comment by adam-narozniak** [comment]
Are you running a Python script or Jupyter Notebook? (In Jupyter Notebook, setting seed before each new run will be necessary if you run one after the other).
